_studio/shared/include/mfx_config.h

```diff
@@ -182,8 +182,16 @@
 	#define MFX_ENABLE_FOURCC_RGB565
 #endif
 
-#if defined(MFX_ENABLE_MPEG2_VIDEO_ENCODE)
+#if defined(MFX_ENABLE_MPEG2_VIDEO_ENCODE) && defined(MFX_VA)
 #define MFX_ENABLE_MPEG2_VIDEO_ENCODE_HW
 #endif
 
+#if defined(AS_H264LA_PLUGIN) || defined(AS_HEVCD_PLUGIN) || defined(AS_HEVCE_PLUGIN) || defined(AS_VP8D_PLUGIN) || defined(AS_VP9D_PLUGIN) || (defined(AS_HEVC_FEI_ENCODE_PLUGIN) && MFX_VERSION >= MFX_VERSION_NEXT)
+#undef MFX_ENABLE_MPEG2_VIDEO_DECODE
+#undef MFX_ENABLE_MPEG2_VIDEO_ENCODE
+#undef MFX_ENABLE_MPEG2_VIDEO_ENCODE_HW
+#undef MFX_ENABLE_MJPEG_VIDEO_DECODE
+#undef MFX_ENABLE_MJPEG_VIDEO_ENCODE
```

Comment:
Do you understand background behind `defined(AS_*_PLUGIN)`? Some plugins linked in part of mediasdk library, these were so called "pseudo-plugins". Since this was _part_ of the library, some links could not be resolved and corresponding features were switched off. Thus, you don't need to list the whole bunch of undefs in the AS_*_PLUGIN section. Instead you need to list only what you really need to switch off due to failure to link the plugin. Guideline: list as little undefs as possible.